system-linux.c

```diff
@@ -1702,55 +1704,85 @@ int system_vlandev_del(struct device *vlandev)
 	return system_link_del(vlandev->ifname);
 }
 
+static void ethtool_link_mode_clear_bit(int nr, __u32 *mask)
+{
+	if (nr < 0)
+		return;
+
+	mask[nr / 32] &= ~(1 << (nr % 32));
```

Comment:
```suggestion
	mask[nr / 32] &= ~(1U << (nr % 32));
```

to avoid undefined behavior for `(nr % 32) == 31`.

Reply:
Ack. done